IPython/core/debugger.py

```diff
@@ -329,8 +336,21 @@ def print_stack_trace(self, context=None):
         except (TypeError, ValueError):
                 raise ValueError("Context must be a positive integer")
         try:
-            for frame_lineno in self.stack:
+            skipped = 0
+            for hidden, frame_lineno in zip(self.hidden_frames(), self.stack):
```

Comment:
Reading that I find it not immediately obvious why `self.hidden_frames()` and `self.stack` should have the same size. Would it be more readable if self.hidden_frames() took a stack in argument? so you would have:
`for hidden, frame_lineno in zip(self.hidden_frames(self.stack), self.stack):`

Reply:
sure, I don't think that changes much. But done.